ksql-serde/src/main/java/io/confluent/ksql/serde/connect/ConnectDataTranslator.java

```diff
@@ -52,60 +58,145 @@ public GenericRow toKsqlRow(final Schema connectSchema,
     );
   }
 
+  private RuntimeException createTypeMismatchException(final String pathStr,
+                                                       final Schema schema,
+                                                       final Schema connectSchema) {
+    throw new DataException(
+        String.format(
+            "Cannot deserialize type %s as type %s for field %s",
+            connectSchema.type().getName(),
+            schema.type().getName(),
+            pathStr));
+  }
+
+  private void validateType(final String pathStr,
+                            final Schema schema,
+                            final Schema connectSchema,
+                            final Schema.Type... validTypes) {
+    Arrays.asList(validTypes).stream()
+        .filter(connectSchema.type()::equals)
+        .findFirst()
+        .orElseThrow(
+            () -> createTypeMismatchException(pathStr, schema, connectSchema));
+  }
+
+  private void validateType(final String pathStr,
```

Comment:
Do we need this method? It is only called from 1 place AFAICT

Reply:
I'll remove it.